zone/water_map.cpp

```diff
@@ -26,17 +18,21 @@ WaterMap* WaterMap::LoadWaterMapfile(std::string zone_name) {
 	std::transform(zone_name.begin(), zone_name.end(), zone_name.begin(), ::tolower);
 
 	std::string filename;
-	if (file_exists("maps")) {
-		filename = "maps";
+	if (FileUtil::exists("maps")) {
+		filename = "maps/";
 	}
-	else if (file_exists("Maps")) {
-		filename = "Maps";
+	else if (FileUtil::exists("Maps")) {
+		filename = "Maps/";
 	}
 	else {
 		filename = Config->MapDir;
+		const auto& dir_len = filename.length();
+		if (dir_len > 0 && filename[dir_len - 1] != '/') {
```

Comment:
This is not cross platform, suggest using `std::filesystem` abstractions to keep path manipulation cross platform.

For example Windows expresses paths via `\` and Linux/Unix `/` 

Example with EQEmu's path manager https://github.com/EQEmu/Server/blob/master/common/path_manager.cpp#L19

Once you have a filesystem object then you can return constructs like so https://en.cppreference.com/w/cpp/filesystem/path (see `root_name` `root_path` etc.)